src/dmd/root/outbuffer.d

```diff
@@ -23,23 +23,60 @@ debug
     debug = stomp; // flush out dangling pointer problems by stomping on unused memory
 }
 
+/**
+`OutBuffer` is a write-only output stream of untyped data. It is backed up by
+a contiguous array or a memory-mapped file.
+*/
 struct OutBuffer
 {
+    import dmd.root.file : FileMapping;
+
+    // IMPORTANT: PLEASE KEEP STATE AND DESTRUCTOR IN SYNC WITH DEFINITION IN ./outbuffer.h.
+    // state {
     private ubyte[] data;
     private size_t offset;
     private bool notlinehead;
-
+    /// File mapping, if any. Use a pointer for ABI compatibility with the C++ counterpart.
+    /// If the pointer is non-null the store is a memory-mapped file, otherwise the store is RAM.
+    private FileMapping!ubyte* fileMapping;
     /// Whether to indent
     bool doindent;
     /// Whether to indent by 4 spaces or by tabs;
     bool spaces;
     /// Current indent level
     int level;
+    // state }
 
+    /**
+    Construct from filename. Will map the file into memory (or create it anew
+    if necessary) and start writing at the beginning of it.
```

Comment:
Please use Ddoc comments. Needs `Params:` section.

Reply:
Thanks. I'll "resolve conversation" for each topic I fix, or add a comment if need be. Then I'll submit the resulting new commit.